rotkehlchen/chain/evm/decoding/balancer/balancer_cache.py

```diff
@@ -0,0 +1,203 @@
+import logging
+from typing import TYPE_CHECKING, Final, Literal
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import TokenEncounterInfo, get_or_create_evm_token
+from rotkehlchen.chain.evm.decoding.balancer.constants import (
+    CPT_BALANCER_V1,
+    CPT_BALANCER_V2,
+)
+from rotkehlchen.chain.evm.decoding.balancer.utils import (
+    query_balancer_pools,
+    query_balancer_pools_count,
+)
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.fval import FVal
+from rotkehlchen.globaldb.cache import (
+    globaldb_get_general_cache_values,
+    globaldb_set_general_cache_values,
+    globaldb_update_cache_last_ts,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address, deserialize_int
+from rotkehlchen.types import CacheType, ChainID, ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_balancer_data(
+        inquirer: 'EvmNodeInquirer',
+        cache_type: Literal[CacheType.BALANCER_V1_POOLS, CacheType.BALANCER_V2_POOLS],
+        msg_aggregator: 'MessagesAggregator',
+) -> tuple[set['ChecksumEvmAddress'], set['ChecksumEvmAddress']]:
+    """Query and store balancer pools with their gauges.
+
+    May raise:
+    - RemoteError
+    """
+    version: Final = 1 if cache_type == CacheType.BALANCER_V1_POOLS else 2
+    protocol = CPT_BALANCER_V1 if version == 1 else CPT_BALANCER_V2
+    latest_pools_count = query_balancer_pools_count(
+        chain=inquirer.chain_id,
+        version=version,
+    )
+    gauge_key_parts = (str(inquirer.chain_id.value), str(version))
+    pool_key_parts = (str(inquirer.chain_id.value),)
```

Comment:
```suggestion
    gauge_key_parts = ((str_chain_id := str(inquirer.chain_id.value)), str(version))
    pool_key_parts = (str_chain_id,)
```

do we have tests that check the keys?

Reply:
yeah, the test in `test_globaldb_cache.py` queries the data and reads from the globaldb cache.